### PR TITLE
[#574] [#505] fix Appveyor bundle and dry Zold::Http

### LINKED ISSUE #505: Dry out #get and #put methods as they have identicall parts
https://github.com/zold-io/zold/blob/e4332805ab78db66eb1157485ba9190048c5e3a0/lib/zold/http.rb#L74-L79

### LINKED ISSUE #574: Appveyor build is broken: libcurl is not found
https://ci.appveyor.com/project/yegor256/zold-jhx6b/builds/20528112/job/y8x1po291wrq3d4v

screen shot 2018-11-26 at 17 14 37

### PR DESCRIPTION
**Pull request by dapi**
1) gem `patreon` has broken windows support that is why I changed it to `typhoeus`. fixes #574 
2) `typhoeus` has well tried syntax’s that is why it fixes #505
3) appveyor is still broken, this PR fixes it - https://github.com/yegor256/futex/pull/14

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#577](https://github.com/zold-io/zold/issues/577) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#577](https://github.com/zold-io/zold/issues/577) is assigned to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa), here is [why](https://www.0crat.com/footprint/CAZPZR9FS/6e5ca362-ef0e-4d15-97f4-3cb606832f77); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by emilianodellacasa** [comment]
@dapi It is always better to have separate PRs for separate issues, this is a main Zerocracy requirement. However, I cannot see any problem in your code, I leave the decision on what to do to @yegor256 , in my opinion we can safely merge

File: test/test_http.rb [hunk 1]
```diff
@@ -84,7 +84,7 @@ def test_terminates_on_timeout
       end
       sleep 0.25
       res = Zold::Http.new(uri: "http://127.0.0.1:#{port}/").get(timeout: 0.1)
-      assert_equal(599, res.status, res)
```
Comment (yegor256) [inline test/test_http.rb]: @dapi this is a serious change in the protocol. We have to get 599 in case of failures. Zero is not an acceptable output.

**Comment by yegor256** [comment]
@dapi indeed, @emilianodellacasa is right, we must not combine solutions under one PR. Also, see my comment above.

File: test/test_http.rb (hunk 1 above)
Comment (dapi) [reply test/test_http.rb]: ok, I get

**Comment by dapi** [comment]
How can I separate it? issue #505 is fixed by side effect ;)

**Comment by dapi** [comment]
@yegor256 look into code. There is one solution that fixes both your issues. Sorry for that, it fixed accidentally.

File: test/test_http.rb (hunk 1 above)
Comment (dapi) [reply test/test_http.rb]: thx

Comment (dapi) [reply test/test_http.rb]: @yegor256 why used this solution? Why does not compare `res` class with `Error` class? Why it does not raise and exception?

Comment (dapi) [reply test/test_http.rb]: I fix that. There are now dependencies on 599 except tests:

```
lib/zold/http.rb:        code == 0 ? 599 : code
lib/zold/http.rb:        599
test/test_http.rb:    assert_equal(599, res.status)
test/test_http.rb:      assert_equal(599, res.status, res)
test/node/fake_node.rb:          break unless ping.status == 599 && node.alive?
```

I think use fictional http status is not good idea. Good:

```
res = http.get()
if res.is_a?(Error)
  do_some()
end
```

Better - raise exception and catch it when you need.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@emilianodellacasa According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

No issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@dapi According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](https://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to

Please correct all your messages and code review comments by indicating to addressee in the beginning.

**Comment by emilianodellacasa** [comment]
@ypshenychka Confirmed

**Comment by ypshenychka** [comment]
@emilianodellacasa thanks

**Comment by ypshenychka** [comment]
@dapi ping

**Comment by ypshenychka** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request closed without merging**